src/components/storage/InstallationScenario.jsx

```diff
@@ -125,13 +129,42 @@ const checkMountPointMapping = ({ mountPointConstraints, selectedDisks, usablePa
     return availability;
 };
 
-// TODO implement
-const checkHomeReuse = ({ selectedDisks }) => {
+const checkHomeReuse = ({ devices, originalExistingSystems, selectedDisks }) => {
     const availability = new AvailabilityState();
 
     availability.hidden = false;
     availability.available = !!selectedDisks.length;
 
+    const isOSOnDisks = (osData, disks) => {
```

Comment:
Can you put this in src/helpers/storage.js? I implemented exactly the same also in another PR, https://github.com/rhinstaller/anaconda-webui/pull/440
So feel free to reuse it?

So something like this in src/helpers/storage
``` js
const getOSonDisk = ({ device }) => { 
    const originalExistingSystems = useOriginalExistingSystems();
    const devices = useOriginalDevices();
    const children = getDeviceChildren({ device, deviceData: devices });
    const existingSystemsOnDevice = originalExistingSystems.filter(
        system => system.devices.v.some(device => children.includes(device))
    );
   return existingSystemsonDevice;
}

```

And then here:
```js
    const isOSOnDisks = (disks) => {
    return disks.some(disk => getOSonDisk({ device: disk }).length > 0);
  }
```

Reply:
My function checks if given OS is residing completely on the disks, while the suggested implementation just checks if on the given disks there is any part of any existing OS.

Reply:
Updated the name of the function